partner_firstname/models/res_partner.py

```diff
@@ -146,9 +162,23 @@ def _get_inverse_name(self, name, is_company=False):
             parts = [name or False, False]
         # Guess name splitting
         else:
-            parts = name.strip().split(" ", 1)
-            while len(parts) < 2:
-                parts.append(False)
+            order = self._get_names_order()
+            # Remove redundant spaces
+            name = re.sub(r'([, ]) +', r'\1', name)
+            if ',' in name:
+                # If a comma found, then use 'Lastname, Firstname' pattern
+                order = 'last_first_comma'
```

Comment:
IMHO you should remove this. This is one of those places where a wrong input should return a wrong result. If database is not configured to use commas, you should not care about them.